src/olympia/blocklist/management/commands/import_blocklist.py

```diff
@@ -14,20 +16,40 @@
 class Command(BaseCommand):
     help = ('Populate AMO blocklist by importing v2 JSON blocklist from kinto')
 
-    KINTO_JSON_BLOCKLIST_URL = 'https://firefox.settings.services.mozilla.com/v1/buckets/blocklists/collections/addons/records'  # noqa
-
     def handle(self, *args, **options):
-        log.debug(
-            'Downloading blocklist from %s', self.KINTO_JSON_BLOCKLIST_URL)
-        response = requests.get(self.KINTO_JSON_BLOCKLIST_URL)
-
+        LEGACY_BLOCKLIST_URL = (
+            settings.REMOTE_SETTINGS_API_URL +
+            'buckets/blocklists/collections/addons/records')
+        log.debug('Downloading blocklist from %s', LEGACY_BLOCKLIST_URL)
+        response = requests.get(LEGACY_BLOCKLIST_URL)
+
+        data = response.json()['data']
+        kinto_ids = [record['id'] for record in data]
         # filter out the blocks we've already imported
-        kinto_ids = [record['id'] for record in response.json()['data']]
-        already_imported = KintoImport.objects.filter(
-            kinto_id__in=kinto_ids).values_list('kinto_id', flat=True)
-        records = [
-            record for record in response.json()['data']
-            if record['id'] not in already_imported]
-        log.debug('%s records from blocklist to process', len(records))
-        for record in records:
+        kinto_import_qs = KintoImport.objects.all().values_list(
+            'kinto_id', 'timestamp', named=True)
+        already_imported = {
+            import_.kinto_id: import_.timestamp for import_ in kinto_import_qs}
+        new_records = [
+            record for record in data
+            if record['id'] not in already_imported
+        ]
+        modified_records = [
+            record for record in data
+            if record['id'] in already_imported and
+            record['last_modified'] != already_imported[record['id']]
+        ]
+        deleted_record_ids = [
+            kinto_id for kinto_id in already_imported
+            if kinto_id not in kinto_ids
+        ]
+        log.debug(
+            '%s new, %s modified, % deleted records from blocklist to process',
+            len(new_records), len(modified_records), len(deleted_record_ids))
+        for record in new_records + modified_records:
             import_block_from_blocklist.delay(record)
+        if deleted_record_ids:
+            log.debug(
+                'Deleting Blocks that have been removed from v2 blocklist')
+            for kinto_id in deleted_record_ids:
+                delete_imported_block_from_blocklist.delay(kinto_id)
```

Comment:
In the issue, there is this paragraph:

> It's important that if something is blocked and then that block is edited/removed later in v2 records (and assuming a manual update to AMO), that the the import won't reverse that manual change (thanks to Philipp for raising this point).

Is this guaranteed?

Reply:
It's not possible for that to happen.  `Block` creations, deletions, and edits don't trigger changes to the v2 collection - creating and signing off a `BlocklistSubmission` does.  We don't create `BlocklistSubmission` instances with this command.

Reply:
* it would if the waffle switch was enabled anyway.  The code behind the waffle switch is obsolete now but we're not removing it until it's clear what's happening in September after the v2 blocklist is deprecated.  With the waffle switch disabled (the default) there are no updates to the v2 blocklist from AMO under any circumstances.